compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -4110,7 +4110,7 @@ object Types {
         cachedSuper = tycon match {
           case tycon: HKTypeLambda => defn.AnyType
           case tycon: TypeRef if tycon.symbol.isClass => tycon
-          case tycon: TypeProxy => tycon.superType.applyIfParameterized(args)
+          case tycon: TypeProxy => tycon.superType.applyIfParameterized(args).normalized
```

Comment:
I am not sure whether the caching is permissible here. If we normalize, are we certain that the next time around when we demand the super type of an applied type, we get the same normalization result? It does not look obvious to me. 
Maybe we should do this instead.
```scala
    override def superType(using Context): Type =
      if ctx.period != validSuper then
        validSuper = if (tycon.isProvisional) Nowhere else ctx.period
        def reappliedSuper = tycon.superType.applyIfParameterized(args).normalized
        cachedSuper = tycon match
          case tycon: HKTypeLambda => defn.AnyType
          case tycon: TypeRef =>
            if tycon.symbol.isClass then tycon
            else
              if tycon.info.isInstanceof[MatchAlias] then validSuper = Nowhere
              reappliedSuper
          case tycon: TypeProxy => reappliedSuper
          case _ => defn.AnyType
      cachedSuper
```

Reply:
I updated the PR to change validity of the cache for normalizing match types.